src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs

```diff
@@ -0,0 +1,529 @@
+
+using System;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using Silk.NET.Core.Contexts;
+using Silk.NET.GLFW;
+using Silk.NET.Windowing.Internals;
+using Silk.NET.Windowing.Structs;
+using VideoMode = Silk.NET.Windowing.VideoMode;
+
+namespace Silk.NET.Windowing.Glfw
+{
+    internal unsafe class GlfwWindow : WindowImplementationBase, IGLContext, IVkSurface
+    {
+        private readonly GLFW.Glfw _glfw;
+        private WindowHandle* _glfwWindow;
+        private string _localTitleCache; // glfw doesn't let us get the window title.
+        private readonly GlfwWindow _parent;
+        private readonly GlfwMonitor _initialMonitor;
+        
+        // Callbacks
+        private GlfwCallbacks.WindowPosCallback _onMove;
+        private GlfwCallbacks.WindowSizeCallback _onResize;
+        private GlfwCallbacks.DropCallback _onFileDrop;
+        private GlfwCallbacks.WindowCloseCallback _onClosing;
+        private GlfwCallbacks.WindowFocusCallback _onFocusChanged;
+        private GlfwCallbacks.WindowIconifyCallback _onMinimized;
+        private GlfwCallbacks.WindowMaximizeCallback _onMaximized;
+
+        public GlfwWindow(WindowOptions optionsCache, GlfwWindow parent, GlfwMonitor monitor) : base(optionsCache)
+        {
+            _glfw = GlfwProvider.GLFW.Value;
+            _parent = parent;
+            _initialMonitor = monitor;
+        }
+
+        protected override Size CoreSize
+        {
+            get
+            {
+                _glfw.GetWindowSize(_glfwWindow, out var width, out var height);
+                return new Size(width, height);
+            }
+        }
+
+        protected override IntPtr CoreHandle => (IntPtr) _glfwWindow;
+
+        protected override void CoreReset()
+        {
+            if (_glfwWindow == null)
+            {
+                return;
+            }
+            
+            try
+            {
+                _glfw.DestroyWindow(_glfwWindow);
+                GLFW.Glfw.ThrowExceptions();
+                _glfwWindow = null;
+            }
+            catch (GlfwException)
+            {
+                // If the window is already destroyed, it throws an exception,
+                // but we want the window destroyed anyways, so just ignore it
+            }
+        }
+
+        public override IGLContext? GLContext
+            => API.API == ContextAPI.OpenGL || API.API == ContextAPI.OpenGLES ? this : null;
+
+        public override IVkSurface? VkSurface => API.API == ContextAPI.Vulkan && _glfw.VulkanSupported() ? this : null;
+        protected override bool CoreIsVisible
+        {
+            get => _glfw.GetWindowAttrib(_glfwWindow, WindowAttributeGetter.Visible);
+            set
+            {
+                if (value)
+                {
+                    _glfw.ShowWindow(_glfwWindow);
+                }
+                else
+                {
+                    _glfw.HideWindow(_glfwWindow);
+                }
```

Comment:
nit: style

Reply:
In the past we haven't really actively sought to use braceless blocks because of the explicitness that braces provide.